### PR TITLE
Closes #537: Refactors private sessions usage

### LINKED ISSUE #537: Refactor private sessions a little
Per a conversation with @boek, private sessions are breaking a few ownership constraints (e.g. HistoryFragment knowing about private session). We can tackle this by just using the selectedSession's state instead of passing around isPrivate.

### PR DESCRIPTION
**Pull request by sblatz**
Note: I am purposefully keeping the usage of 
`(activity as HomeActivity).browsingModeManager.isPrivate` in HomeFragment as it is possible the user will not have a selected session and I don't want to crash (simply setting to false if null doesn't work either as we're persisting private mode).

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 1]
```diff
@@ -59,7 +58,7 @@ class HistoryFragment : Fragment(), CoroutineScope, BackHandler {
         Navigation.findNavController(requireActivity(), R.id.container).apply {
             navigate(
                 HistoryFragmentDirections.actionGlobalBrowser(null,
-                    (activity as HomeActivity).browsingModeManager.isPrivate),
+                    requireComponents.core.sessionManager.selectedSession!!.private),
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt]: I would never assume there's a selected session. It's definitely not true on custom tabs. It's likely not true when the user first enters the app.

File: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt [hunk 2]
```diff
@@ -88,7 +88,7 @@ class SearchFragment : Fragment() {
     private fun transitionToBrowser() {
         val sessionId = SearchFragmentArgs.fromBundle(arguments!!).sessionId
         val directions = SearchFragmentDirections.actionSearchFragmentToBrowserFragment(sessionId,
-            (activity as HomeActivity).browsingModeManager.isPrivate)
+            requireComponents.core.sessionManager.selectedSession!!.private)
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt]: Again, this is an unsafe assumption. There is not always a selected session.

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt]: Also: If you really really are sure (maybe never in Fenix?) then you can also use [SessionManager.selectedSessionOrThrow](https://mozac.org/api/mozilla.components.browser.session/-session-manager/selected-session-or-throw.html)

**Comment by sblatz** [comment]
Closing this in favor of creating a solution that holds the "app state" so we don't have to rely on the selected session.

**Pull request closed without merging**